go-controller/pkg/node/rulemanager/rule_manager.go

```diff
@@ -0,0 +1,182 @@
+package rulemanager
+
+import (
+	"sync"
+	"time"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+
+	"github.com/vishvananda/netlink"
+)
+
+type nlRule struct {
+	rule   *netlink.Rule
+	delete bool
+}
+
+type Controller struct {
+	mu            *sync.Mutex
+	rules         []nlRule
+	ownPriorities map[int]bool
+	v4            bool
+	v6            bool
+}
+
+// NewController creates a new linux rule manager
+func NewController(v4, v6 bool) *Controller {
+	return &Controller{
+		mu:            &sync.Mutex{},
+		rules:         make([]nlRule, 0),
+		ownPriorities: make(map[int]bool, 0),
```

Comment:
probably need some description on what this is for. I'm guessing keeping track of the priority numbers we use?

Reply:
updated description - owning a priority means only explicit rules for that priority are allowed - anything else is removed.